core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -337,7 +337,7 @@ open class TileInfo {
         }
         lineList += baseTerrain.tr()
         if (terrainFeature != null) lineList += terrainFeature!!.tr()
-        if (viewingCiv==null || hasViewableResource(viewingCiv)) lineList += resource!!.tr()
+        if (viewingCiv==null && resource!=null || viewingCiv!=null && hasViewableResource(viewingCiv)) lineList += resource!!.tr()
```

Comment:
With nested boolean logic it's better to have (brackets) so the order is clearer.
In this case, I see what the problem was - but I think it should be
`if(resource!=null && (viewingCiv==null || hasViewableResouce(viewingCiv))`

Reply:
That's already 10 times the effort I spent so far - which was 0.5s code and 5s comment.